### PR TITLE
Enabled Eigen preconditioner configuration

### PR DESCRIPTION
**Pull request by chleh**
Until now the setting in the prj file was not strictly parsed and ignored later on.

### CODE REVIEW

**Comment by norihiro-w** [comment]
looks good

**Comment by chleh** [comment]
Forgot to mention: The prj file for our ctests have to change: ufz/ogs-data#21

File: MathLib/LinAlg/Eigen/EigenLinearSolver.cpp [hunk 1]
```diff
@@ -115,25 +162,16 @@ EigenLinearSolver::EigenLinearSolver(
...
-        break;
-    }
-    case EigenOption::SolverType::CG: {
-        using SolverType = Eigen::ConjugateGradient<Matrix, Eigen::Lower, Eigen::DiagonalPreconditioner<double>>;
-        _solver.reset(new details::EigenIterativeLinearSolver<SolverType>);
-        break;
-    }
-    case EigenOption::SolverType::INVALID:
-        OGS_FATAL("Invalid Eigen linear solver type. Aborting.");
+    switch (_option.solver_type) {
```
Comment (endJunction) [inline MathLib/LinAlg/Eigen/EigenLinearSolver.cpp]: Switch with two cases is an if-else...
Is the todo comment above still correct? I mean there is (non-obvious) OGS_FATAL in createIterativeSolver...

**Comment by endJunction** [comment]
I'm not into Eigen3 now, but I think for SparseLU there was no preconditioner, only for iterative solvers.
Ignore this comment if it makes no sense.

**Comment by endJunction** [comment]
:fast_forward:

File: MathLib/LinAlg/Eigen/EigenLinearSolver.cpp (hunk 1 above)
Comment (chleh) [inline MathLib/LinAlg/Eigen/EigenLinearSolver.cpp]: Fixed the switch.
The todo is still valid: The default solver is set in EigenOption.cpp; pretty hidden, I think.

**Comment by chleh** [comment]
>  I think for SparseLU there was no preconditioner

For SparseLU the preconditioner setting is ignored.

**Comment by chleh** [comment]
Failing tests not related to this PR. Going to merge.

**Pull request merged**